Introduce single 'Set up iOS development' page (#11963)
flutter/website: src/content/platform-integration/ios/setup-ios.md

Reviewer: What do you think about the file just being `setup.md`? I feel the `ios` is clear from the parent directory.

---
title: Set up iOS development
description: >-
  Configure your development environment to
  run, build, and deploy Flutter apps for iOS devices.
short-title: Set up iOS
---

Learn how to set up your development environment
to run, build, and deploy Flutter apps for iOS devices.

> 

Reviewer 2: This is missing the Rosetta instructions for Apple Silicon (ARM) Macs:
```
sudo softwareupdate --install-rosetta --agree-to-license
```
https://github.com/flutter/website/pull/7119#issuecomment-1124537969
https://github.com/flutter/flutter/issues/69157 (getting closer to done)

:::note
If you haven't set up Flutter already,
visit and follow the [Get started with Flutter][] guide first.

If you have already installed Flutter,
ensure that it's [up to date][].
:::

[Get started with Flutter]: /get-started/install
[up to date]: /install/upgrade

> [up to date]: /install/upgrade

Reviewer 3: FYI this is repeated from line 21

## Install Xcode

With Xcode, you can run Flutter apps on
an iOS physical device or on the iOS Simulator.

If you haven't done so already,
install and set up [Xcode][].
If you have installed Xcode, ensure that
it's up to date and continue to the next step.
(After setting up the command-line tools
and accepting the license, as shown here.)

[Xcode]: https://developer.apple.com/xcode/

> To configure Xcode's command-line tools,
> use the following command:

Reviewer 3: What do you think of this?

```suggestion
Configure Xcode's command-line tools:
```

```console
$ sudo sh -c 'xcode-select -s /Applications/Xcode.app/Contents/Developer && xcodebuild -runFirstLaunch'

> ```

Reviewer 2: This part doesn't require sudo, but `xcodebuild -license` does. The three commands are:
```
$ xcode-select -s /Applications/Xcode.app
$ xcodebuild -runFirstLaunch
$ sudo xcodebuild -license
```
I also think `xcodebuild -downloadPlatform iOS` is required for both physical devices and simulator, so you can add it to the Xcode section instead of in the "Virtual device" section.

Sign the Xcode license agreement:

```console
$ sudo xcodebuild -license
```

## Configure your target iOS device

Note that the behavior of the iOS Simulator can vary
from an actual physical device. It's easier to get
the Simulator up and running than a physical device,
which requires more setup.
For that reason, we recommend starting with the Simulator.

{% tabs "ios-simulator-or-physical-device" %}
{% tab "Simulator" %}

> ### Configure your iOS Simulator

Reviewer 2: I have never needed to do a single step in the "Configure your iOS Simulator" section. It was added in https://github.com/flutter/website/pull/9054 which links to https://github.com/flutter/website/issues/8990, which seems unrelated...
@vashworth any idea why we would need to keep the "use a 64-bit device" or "high-screen density" parts?

Current page is:
https://docs.flutter.dev/get-started/install/macos/mobile-ios#configure-your-ios-simulator

This PR is staged at https://flutter-docs-prod--pr11963-get-st-9rrn1yhn.web.app/platform-integration/ios/setup-ios

Use the following steps:

> 1. To install the iOS Simulator, run the following command.

Reviewer 3: Consider:

```suggestion
1. To install the iOS Simulator, run the following command:
```

Alternatively:

```suggestion
1. Install the iOS Simulator:
```

    | HD accurate       | **Window** <span aria-label="and then">></span> **Pixel Accurate** | <kbd>Cmd</kbd> + <kbd>3</kbd> |
    | Fit to screen     | **Window** <span aria-label="and then">></span> **Fit Screen**     | <kbd>Cmd</kbd> + <kbd>4</kbd> |
    
    {:.table .table-striped}

{% endtab %}
{% tab "Physical device" %}

Set up each iOS device on which you want to test.

### Configure your physical iOS device {:#attach}

1. Attach your iOS device to the USB port on your Mac.

1. On first connecting an iOS device to your Mac,
   your device will display the **Trust this computer?** dialog.

1. Click **Trust**.

   ![Trust Mac][]

### Enable Developer Mode

Apple requires enabling **[Developer Mode][]**
on the device to protect against malicious software.

1. Tap on **Settings** <span aria-label="and then">></span>
   **Privacy & Security** <span aria-label="and then">></span>
   **Developer Mode**.

1. Tap to toggle **Developer Mode** to **On**.

1. Restart the device.

1. When the **Turn on Developer Mode?** dialog appears,
   tap **Turn On**.

   The dialog explains that Developer Mode requires reducing

>    the security of the iOS device.

Reviewer 3: Should we remove this? It feels a bit self-explanatory. It's also quite similar to this section's intro on line 135.

are just _testing_ your app on an iOS device,
a personal Apple Developer account is free and works.

:::note Apple Developer program
When you want to _deploy_ your app to the
App Store, you will need to upgrade your
personal Apple Developer account to
a professional account.
:::

### Prepare the device

Find your **Profiles & Device Management** menu under
**Settings**. (The exact location can vary by device.)
Toggle your certificate to **Enable**.

:::note
If you can't find **VPN & Device Management** menu,
run your app on your iOS device once, then try again.
:::

1. Under the **Developer App** heading,
   you should find your certificate.

1. Tap the certificate.

1. Tap **Trust "&lt;certificate&gt;"**.

1. When the dialog displays, tap **Trust**.

If the **codesign wants to access key...** dialog appears:

1. Enter your macOS password.

1. Tap **Always Allow**.

[Trust Mac]: /assets/images/docs/setup/trust-computer.png
[Developer Mode]: {{site.apple-dev}}/documentation/xcode/enabling-developer-mode-on-a-device
[Apple Developer]: {{site.apple-dev}}/programs/
[Apple Developer account]: {{site.apple-dev}}/account

{% endtab %}
{% endtabs %}

---

> ## Install CocoaPods

Reviewer 3: Should we move the CocoaPods section to be before the iOS simulator / physical device section? That order might feel more natural: I configure my Mac then I configure my target device. Currently these are mixed since the CocoaPods step is after the target device section.

If your app depends on any [Flutter plugin][]
that uses native iOS code,
install [CocoaPods][cocoapods].
CocoaPods bundles various dependencies that
support connecting Flutter to native iOS code.

1. Install `cocoapods` following the
   [CocoaPods install guide][cocoapods].

   ```console
   $ sudo gem install cocoapods
   ```

1. Launch your preferred text editor to update its path variable.

1. For example, if you use Zsh, open `~/.zshenv` in your text editor.

1. Copy the following line and paste it at the end of your `~/.zshenv` file.

   ```bash
   export PATH=$HOME/.gem/bin:$PATH
   ```

1. Save your `~/.zshenv` file.

> 1. To apply this change, restart all open terminal sessions.

Reviewer 2: I'm not sure why `.gem/bin` would be needed on your `PATH`, I don't have this set, and it's not in the CocoaPods instructions...
cc @vashworth again to fact check me.

[cocoapods]: https://guides.cocoapods.org/using/getting-started.html#installation
[Flutter plugin]: /packages-and-plugins/developing-packages#types

## Validate your development setup

The `flutter doctor` command validates all required components of a
complete Flutter development environment.

From the command-line, run the following command:

   ```console
   $ flutter doctor
   ```

Since these instructions did not install for Android
or desktop development, you can ignore Flutter doctor's
error reports for those components for now.

If the `flutter doctor` command returns other errors,
run it again with the verbose flag.

```console
$ flutter doctor -v
```

Follow the instructions in the output for software
you might need to install or further tasks to perform.

**Congratulations.**

> You can now start developing Flutter apps for iOS!

Reviewer 3: I love this cheery ending :)

---

> title: Install and set up for iOS development

Reviewer: Consider simplifying this. I think "set up" can include any necessary installation.

```suggestion
title: Set up iOS development
```

> description: Configure your system to develop Flutter for iOS.

Reviewer: ```suggestion
description: >-
  Configure your development environment to
  run, build, and deploy Flutter apps for iOS devices.
```

short-title: Set up iOS
---

> ## Install Flutter
> 
> To set up your development environment for iOS,
> you first need to install Flutter on your
> development machine.
> 
> If you have already installed Flutter,
> ensure that it's [up to date][].
> 
> :::note Did you know?
> When you install or update the Flutter framework,
> it also installs and updates the Dart language for you.
> :::
> 
> If Flutter is installed and up to date,
> continue to the next step.

Reviewer: What do you think about reducing this section a bit? I believe most users navigating to this page should already have installed Flutter or even directly navigating to it from the Flutter set up page.

Combining yours with what I have for the WIP Android page:

```suggestion
Learn how to set up your development environment
to run, build, and deploy Flutter apps for iOS devices.

:::note
If you haven't set up Flutter already,
visit and follow the [Get started with Flutter][] guide first.

If you have already installed Flutter,
ensure that it's [up to date][].
:::

[Get started with Flutter]: /get-started/install
[up to date]: /install/upgrade
```

install and set up [Xcode][].
If you have installed Xcode, ensure that
it's up to date and continue to the next step.
(After setting up the command-line tools
and accepting the license, as shown here.)

[Xcode]: https://developer.apple.com/xcode/

To configure Xcode's command-line tools,
use the following command:

```console
$ sudo sh -c 'xcode-select -s /Applications/Xcode.app/Contents/Developer && xcodebuild -runFirstLaunch'
```

Sign the Xcode license agreement:

```console
$ sudo xcodebuild -license
```

## Configure your target iOS device

Note that the behavior of the iOS Simulator can vary
from an actual physical device. It's easier to get
the Simulator up and running than a physical device,
which requires more setup.
For that reason, we recommend starting with the Simulator.

{% tabs "ios-simulator-or-physical-device" %}
{% tab "Simulator" %}

### Configure your iOS Simulator

Use the following steps:

1. To install the iOS Simulator, run the following command.

    ```console
    $ xcodebuild -downloadPlatform iOS
    ```

1. Start the Simulator with the following command:

    ```console
    $ open -a Simulator
    ```

1. Set your Simulator to use a 64-bit device.

>    This covers iPhone 5s or later.

Reviewer: I think you can drop this note. I believe we don't support any devices before the iPhone 5s.

```suggestion
```

### Enable Developer Mode

Apple requires enabling **[Developer Mode][]**
on the device to protect against malicious software.

1. Tap on **Settings** <span aria-label="and then">></span>
   **Privacy & Security** <span aria-label="and then">></span>
   **Developer Mode**.

1. Tap to toggle **Developer Mode** to **On**.

1. Restart the device.

1. When the **Turn on Developer Mode?** dialog appears,
   tap **Turn On**.

   The dialog explains that Developer Mode requires reducing
   the security of the iOS device.

### Create a developer code signing certificate

To send your app to a physical iOS device,
_even_ for testing, you must establish trust
between your Mac and the device.
In addition to trusting the device when that
popup appears, you must upload a signed
developer certificate to your device.

To create a signed development certificate,
you need an Apple ID.
If you don't have one, create one.
You must also enroll in the [Apple Developer][] program
and create an [Apple Developer account][]. If you
are just _testing_ your app on an iOS device,
a personal Apple Developer account is free and works.

:::note Apple Developer program
When you want to _deploy_ your app to the

> Play store, you will need to upgrade your

Reviewer: ```suggestion
App Store, you will need to upgrade your
```

1. Tap **Always Allow**.

[Trust Mac]: /assets/images/docs/setup/trust-computer.png
[Developer Mode]: {{site.apple-dev}}/documentation/xcode/enabling-developer-mode-on-a-device
[Apple Developer]: {{site.apple-dev}}/programs/
[Apple Developer account]: {{site.apple-dev}}/account

{% endtab %}
{% endtabs %}

---

## Install CocoaPods

If your app depends on any [Flutter plugin][]
that uses native iOS code,
install [CocoaPods][cocoapods].
CocoaPods bundles various dependencies that
support connecting Flutter to native iOS code.

1. Install `cocoapods` following the
   [CocoaPods install guide][cocoapods].

   ```console
   $ sudo gem install cocoapods
   ```

1. Launch your preferred text editor to update its path variable.

1. For example, if you use Zsh, open `~/.zshenv` in your text editor.

1. Copy the following line and paste it at the end of your `~/.zshenv` file.

   ```bash
   export PATH=$HOME/.gem/bin:$PATH
   ```

1. Save your `~/.zshenv` file.

1. To apply this change, restart all open terminal sessions.

[cocoapods]: https://guides.cocoapods.org/using/getting-started.html#installation
[Flutter plugin]: /packages-and-plugins/developing-packages#types

> ## Check your development setup with Flutter doctor

Reviewer: For an easier to scan title, I think can drop the mention of the specific tool as the content mentions it:

```suggestion
## Validate your development setup
```

The `flutter doctor` command validates all required components of a
complete Flutter development environment.

From the command-line, run the following command:

   ```console
   $ flutter doctor
   ```

Since these instructions did not install for Android
or desktop development, you can ignore Flutter doctor's
error reports for those components for now.

If the `flutter doctor` command returns other errors,
run it again with the verbose flag.

```console
$ flutter doctor -v

> ```

Reviewer: Maybe we could simplify this by calling out specifically entries to look for in terms of iOS development.

Perhaps the Xcode entry of `flutter doctor` and for the existence of an iOS devices in `flutter devices`?

Follow the instructions in the output for software
you might need to install or further tasks to perform.

**Congratulations.**

> You can now start developing Flutter apps for iOS!

Reviewer: I have some infrastructure work planned to improve this next step section for the Android version of this page. Once we land that, we can update this section to use it as well.
